rules/library.bzl

```diff
@@ -841,62 +841,47 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
 
     ## BEGIN HMAP
 
-    public_hmap_name = name + "_public_hmap"
-    public_hdrs_filegroup = name + "_public_hdrs"
-    native.filegroup(
-        name = public_hdrs_filegroup,
-        srcs = objc_hdrs,
-        tags = _MANUAL,
-    )
+    if len(objc_hdrs) > 0:
+        public_hmap_name = name + "_public_hmap"
+        public_hdrs_filegroup = name + "_public_hdrs"
+        native.filegroup(
+            name = public_hdrs_filegroup,
+            srcs = objc_hdrs,
+            tags = _MANUAL,
+        )
 
-    # Public hmaps are for vendored static libs to export their header only.
-    # Other dependencies' headermaps will be generated by li_ios_framework
-    # rules.
-    headermap(
-        name = public_hmap_name,
-        namespace = namespace,
-        hdrs = [public_hdrs_filegroup],
-        tags = _MANUAL,
-    )
-    private_dep_names.append(public_hmap_name)
+        # Public hmaps are for vendored static libs to export their header only.
+        # Other dependencies' headermaps will be generated by li_ios_framework
+        # rules.
+        headermap(
+            name = public_hmap_name,
+            namespace = namespace,
+            hdrs = [public_hdrs_filegroup],
+            tags = _MANUAL,
+        )
+        private_dep_names.append(public_hmap_name)
+        _append_headermap_copts(public_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
 
-    private_hmap_name = name + "_private_hmap"
-    private_angled_hmap_name = name + "_private_angled_hmap"
-    private_hdrs_filegroup = name + "_private_hdrs"
-    private_angled_hdrs_filegroup = name + "_private_angled_hdrs"
+    if len(objc_non_exported_hdrs + objc_private_hdrs) > 0:
+        private_hmap_name = name + "_private_hmap"
+        private_hdrs_filegroup = name + "_private_hdrs"
 
-    native.filegroup(
-        name = private_hdrs_filegroup,
-        srcs = objc_non_exported_hdrs + objc_private_hdrs + objc_hdrs,
-        tags = _MANUAL,
-    )
-    native.filegroup(
-        name = private_angled_hdrs_filegroup,
-        srcs = objc_non_exported_hdrs + objc_private_hdrs,
-        tags = _MANUAL,
-    )
+        native.filegroup(
+            name = private_hdrs_filegroup,
+            srcs = objc_non_exported_hdrs + objc_private_hdrs,
```

Comment:
was dropping `objc_hdrs` from here intentional?

Reply:
Won't it just get it from `public_hdrs_filegroup` now?

Reply:
Yes, `objc_hdrs` was added to the `public_hdrs_filegroup` above.  I don't think we need to add it to the private one again.

Reply:
Ah, I missed that... thanks!